api/src/main/java/com/datastrato/gravitino/rel/expressions/FunctionExpression.java

```diff
@@ -73,7 +74,17 @@ public String toString() {
       if (arguments.length == 0) {
         return functionName + "()";
       }
-      return functionName + "(" + String.join(", ", Arrays.toString(arguments)) + ")";
+      String[] functionArguments =
```

Comment:
for example, change `bucket([10, id])` to `bucket(10, id)`

Reply:
seems reasonable, @mchades please check it again

Reply:
@caican00 Could you please add UT to cover this change?

Reply:
> @caican00 Could you please add UT to cover this change?

ok, i will fix it.

Reply:
Sorry, after thinking for a while, it seems unnecessary to change this `toString` method because the method is for developers to read rather than SQL use. The current change will lose the data type information of Literal.

Reply:
> Sorry, after thinking for a while, it seems unnecessary to change this `toString` method because the method is for developers to read rather than SQL use. The current change will lose the data type information of Literal.

`functionName + "(" + String.join(", ", Arrays.toString(arguments)) + ")";`

@mchades  thanks for your reply, but I'm still a little confused. compared to the current changes, the previous also converted to string and lost the data type information of Literal, but the current changes makes the function expression much easier to understand.

Reply:
For example, `Arrays.toString(arguments) `makes the string result of the bucket function `bucket([10, id])` instead of `bucket(10, id)`. and `bucket(10, id) `is closer to the representation of a function.

Reply:
Moreover, we support iceberg's reserved properties later, and need to rely on this `toString` method to represent `sort-order` of Iceberg.

Reply:
For example, as function `bucket(3, col_1)`, the toString result of:
previous: `bucket([LiteralImpl{value=3, dataType=com.datastrato.gravitino.rel.types.Types$IntegerType@3db663d0}, col_1])`, we can know that literal `3` is `IntegerType`, not other numeric type

current change: `bucket(3, col_1)`

Reply:
> Moreover, we support iceberg's reserved properties later, and need to rely on this `toString` method to represent `sort-order` of Iceberg.

Why do Iceberg's reserved properties rely on the `toString` method? Can you give an example? 
This toString rule should be implemented in the Iceberg catalog?

Reply:
> For example, as function `bucket(3, col_1)`, the toString result of:
> 
> * previous: `bucket([LiteralImpl{value=3, dataType=com.datastrato.gravitino.rel.types.Types$IntegerType@3db663d0}, col_1])`, we can know that literal `3` is `IntegerType`, not other numeric type
> * current change: `bucket(3, col_1)`, miss the data type of `3`

bucket: https://github.com/datastrato/gravitino/blob/03c8e2f2a754751638ca95ad2408cd50d3de723e/catalogs/catalog-lakehouse-iceberg/src/main/java/com/datastrato/gravitino/catalog/lakehouse/iceberg/converter/ToIcebergSortOrder.java#L60-L62

truncate: https://github.com/datastrato/gravitino/blob/03c8e2f2a754751638ca95ad2408cd50d3de723e/catalogs/catalog-lakehouse-iceberg/src/main/java/com/datastrato/gravitino/catalog/lakehouse/iceberg/converter/ToIcebergSortOrder.java#L79-L81

@mchades for these two functions, we know the dataType of their first argument should be Integer, it seems no need to show in `toString` method?

Reply:
For the Iceberg catalog, this is indeed clear. 
But for others, when you encounter a string value `20240521`, can you determine if it is of type `date` or `int`?

Reply:
> > Moreover, we support iceberg's reserved properties later, and need to rely on this `toString` method to represent `sort-order` of Iceberg.
> 
> Why do Iceberg's reserved properties rely on the `toString` method? Can you give an example? This toString rule should be implemented in the Iceberg catalog?

I would like to add reserved properties to the properties of `IcebergTable` in Gravitino, and i would then be able to get the reserved properties directly from the Gravitino `IcebergTable` in the `spark-connector`.
In particular, I want the string result of adding the `sort-order` property to be a normal function expression, such as `bucket(10, col1)`.

For example, when we call `sortOrders[i].expression()`, we can get a `bucket(10, col1)` directly instead of `bucket([LiteralImpl{value=3, dataType=com.datastrato.gravitino.rel.types.Types$IntegerType@3db663d0}, col_1])` or bucket([10, col1]),  and then we can add the `bucket(10, col1)` directly to the `sort-order` property.

https://github.com/datastrato/gravitino/blob/c94eb48fbd0cdeb1f27b6d1647e12c64ee3942c4/catalogs/catalog-lakehouse-iceberg/src/test/java/com/datastrato/gravitino/catalog/lakehouse/iceberg/integration/test/CatalogIcebergBaseIT.java#L1229

Reply:
> For the Iceberg catalog, this is indeed clear. But for others, when you encounter a string value `20240521`, can you determine if it is of type `date` or `int`?

ok, i got it. i will support iceberg reserved properties by other way and will roll this back.

Reply:
> > For the Iceberg catalog, this is indeed clear. But for others, when you encounter a string value `20240521`, can you determine if it is of type `date` or `int`?
> 
> ok, i got it. i will support iceberg reserved properties by other way and will roll this back.

sorry about this, after thinking for a while, i think that we can directly determine its type by string value, if it is `date` type, it should be `2024-05-21`, not `20240521`.

Do we have scenes about we must display the type? cc: @mchades

Reply:
> > > For the Iceberg catalog, this is indeed clear. But for others, when you encounter a string value `20240521`, can you determine if it is of type `date` or `int`?
> > 
> > 
> > ok, i got it. i will support iceberg reserved properties by other way and will roll this back.
> 
> sorry about this, after thinking for a while, i think that we can directly determine its type by string value, if it is `date` type, it should be `2024-05-21`, not `20240521`.
> 
> Do we have scenes about we must display the type? Or what engines must display the argument types of functions? cc: @mchades

Hi @mchades may i ask that should i really roll back this change?

Reply:
After consideration, I decided to roll back this change, iceberg reserved properties actually are not strongly dependent on this change